test/testother.cpp

```diff
@@ -1551,14 +1551,15 @@ class TestOther : public TestFixture {
         ASSERT_EQUALS("", errout.str());
 
         check("class X {\n"
-              "    uint64_t i;\n"
+              "    void* i;\n"
               "};\n"
               "class Y : X {\n"
-              "    uint64_t j;\n"
+              "    void* j;\n"
+              "    void* k;\n"
               "};\n"
               "void f(X x, Y y) {\n"
               "}");
-        ASSERT_EQUALS("[test.cpp:7]: (performance) Function parameter 'y' should be passed by reference.\n", errout.str());
+        ASSERT_EQUALS("[test.cpp:8]: (performance) Function parameter 'y' should be passed by reference.\n", errout.str());
```

Comment:
As I understand, that test was checking that size calculations correctly handle inheritance. With that in mind I have adjusted it to use new unit of size (machine word a.k.a sizeof_pointer instead of bytes) that is used in passedByValue check.

Why I changed existing test instead of adding a new one? That is because the old test would fail
on 64-bit platforms. This change effectively increased the size limit from 8 to 16 bytes on 64-bit platforms, but the old test was using absolute sizes and `2*sizeof(uint64_t)` is still not big enough by the new logic. So, keeping the old test as is isn't an option.

As for new tests, I could add tests that would check that passing `struct X { void* p1; void* p2; };` by value is ok and passing `struct Y { void* p1; void* p2; char c; };` isn't.